### PR TITLE
BUG: FormHandler updates meta should be serialized via CustomJSONEncoder #62

### LINKED ISSUE #62: FormHandler edits fail ID column with integer type
FormHandler edits fail ID column with integer type

For below `gramex.yaml` setup

```yaml
url:
  d1:
    pattern: /$YAMLURL/d1/
    handler: FormHandler
    kwargs:
      url: $YAMLPATH/data.csv
      id: id
```

`data.csv` as

```
id,a,b,c
1,3,b,c
```

Throws this error

```
INFO    06-Feb 16:45:13 __init__ 9988 200 GET /d1/?_limit=20&_format=json&_meta=y (127.0.0.1) 17.00ms d1
meta AttrDict([('data', AttrDict([('filters', [('id', '', (1,))]), ('ignored', []), ('count', 1)]))])
ERROR   06-Feb 16:45:33 web Uncaught exception PUT /d1/ (127.0.0.1)
HTTPServerRequest(protocol='http', host='127.0.0.1:9988', method='PUT', uri='/d1/', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "D:\apps\Anaconda3\lib\site-packages\tornado\web.py", line 1592, in _execute
    result = yield result
  File "D:\apps\Anaconda3\lib\site-packages\tornado\gen.py", line 1133, in run
    value = future.result()
  File "D:\apps\Anaconda3\lib\site-packages\tornado\gen.py", line 1141, in run
    yielded = self.gen.throw(*exc_info)
  File "d:\gitgram\gramex\gramex\handlers\formhandler.py", line 230, in put
    yield self.update(gramex.data.update, *path_args, **path_kwargs)
  File "D:\apps\Anaconda3\lib\site-packages\tornado\gen.py", line 1133, in run
    value = future.result()
  File "D:\apps\Anaconda3\lib\site-packages\tornado\gen.py", line 307, in wrapper
    result = func(*args, **kwargs)
  File "D:\apps\Anaconda3\lib\types.py", line 248, in wrapped
    coro = func(*args, **kwargs)
  File "d:\gitgram\gramex\gramex\handlers\formhandler.py", line 218, in update
    self.write(json.dumps(meta, indent=2))
  File "D:\apps\Anaconda3\lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 201, in encode
    chunks = list(chunks)
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 430, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 404, in _iterencode_dict
    yield from chunks
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 325, in _iterencode_list
    yield from chunks
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 437, in _iterencode
    o = _default(o)
  File "D:\apps\Anaconda3\lib\json\encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'int64' is not JSON serializable
```

Tested against dev branch.

### PR DESCRIPTION
**Pull request by pratapvardhan**
Resolves #62

### CODE REVIEW

**Comment by sanand0** [comment]
@pratapvardhan agreed on the change. I'd like to set up CI before we merge this, please.

**Comment by sanand0** [comment]
Closed via 3cb7e67a

Validated via https://code.gramener.com/cto/gramex/pipelines/75942

File: tests/test_formhandler.py [hunk 1]
```diff
@@ -601,6 +601,14 @@ def test_date_comparison(self):
             expected.index = actual.index
             afe(actual, expected, check_like=True)
 
+    def test_edit_id_type(self):
+        target = self.copy_file('sales.xlsx', 'sales-edits.xlsx')
+        tempfiles[target] = target
+        args = {'sales': [1], 'date': [0]}
```
Comment (pratapvardhan) [inline tests/test_formhandler.py]: Should be valid date, else due to recent fix on inferring types, will make this test fail. Will patch this with

```
args = {'sales': [1], 'date': ['2018-01-10']}
```

Comment (sanand0) [reply tests/test_formhandler.py]: OK. Thanks @pratapvardhan -- please send a PR before %1.51

**Pull request closed without merging**